packages/design-system-scripts/gulp/docs/generatePages/generatePages.js

```diff
@@ -73,27 +90,47 @@ function generatedPagesCount(resultGroups) {
   return count;
 }
 
+// Use `changedPath` to only process pages detected from gulp watch
```

Comment:
We use the path of the changed file to selectively filter out files that didnt change, reducing the amount of pages that get regenerated during `yarn start`